Explorer/Assets/DCL/Interaction/Systems/ProcessPointerEventsSystem.cs

```diff
@@ -0,0 +1,238 @@
+                        Material materialToUse = isAtDistance ? hoverMaterial : hoverOorMaterial;
+
+                        if (!materialOnUseByEntity.ContainsKey(entityRef))
+                        {
+                            materialOnUseByEntity.TryAdd(entityRef, materialToUse);
+                            originalMaterialsByEntity.TryAdd(entityRef, new Dictionary<EntityReference, Material[]>());
+
+                            TryAddHoverMaterialToComponentSiblings(materialToUse, world, entityRef);
+                        }
+                        else
+                        {
+                            if (materialOnUseByEntity[entityRef] != materialToUse)
+                            {
+                                // remove material, its going to be properly added next frame
+                                TryRemoveHoverMaterials(entityInfo);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (candidateForHoverLeaveIsValid)
+            {
+                hoverStateComponent.IsHoverOver = true;
+
+                TryRemoveHoverMaterials(previousEntityInfo);
+
+                HoverFeedbackUtils.TryIssueLeaveHoverEventForPreviousEntity(in raycastResult, in previousEntityInfo);
+            }
+        }
+
+        private void TryRemoveHoverMaterials(GlobalColliderEntityInfo entityInfo)
+        {
+            World world = entityInfo.EcsExecutor.World;
+            EntityReference entity = entityInfo.ColliderEntityInfo.EntityReference;
+
+            if (materialOnUseByEntity.ContainsKey(entity))
+            {
+                using (entityInfo.EcsExecutor.Sync.GetScope())
+                {
+                    if (entity.IsAlive(world)) { TryRemoveHoverMaterialFromComponentSiblings(world, entity); }
+                }
+
+                materialOnUseByEntity.Remove(entity);
+            }
+        }
+
+        private void TryRemoveHoverMaterialFromComponentSiblings(World world, EntityReference entityRef)
+        {
+            if (!world.TryGet(entityRef, out TransformComponent transformComponent)) return;
+            if (!world.TryGet(transformComponent.Parent, out TransformComponent parentTransform)) return;
+
+            Dictionary<EntityReference, Material[]> materialDict = originalMaterialsByEntity[entityRef];
+
+            foreach (EntityReference brother in parentTransform.Children)
+            {
+                if (!world.TryGet(brother, out PrimitiveMeshRendererComponent primitiveMeshRendererComponent))
+                    continue;
+
+                if (!materialDict.ContainsKey(brother)) continue;
+
+                primitiveMeshRendererComponent.MeshRenderer.sharedMaterials = materialDict[brother];
+                materialDict.Remove(brother);
+            }
+        }
+
+        private void TryAddHoverMaterialToComponentSiblings(Material targetMaterial, World world, EntityReference entityRef)
```

Comment:
I think the logic for adding and removing hover material should be moved from the global to scene world (to individual scene instead of operating with the foreign world and `EntityReference`)